File: src/main/java/com/thebluehats/server/game/managers/world/PerkManager.java
package com.thebluehats.server.game.managers.world;

import java.util.ArrayList;
import java.util.Arrays;

import com.thebluehats.server.game.perks.Perk;
import com.thebluehats.server.game.utils.Registerer;

public class PerkManager implements Registerer<Perk> {
    private final ArrayList<Perk> perks = new ArrayList<>();

    @Override
    public void register(Perk[] perks) {
        this.perks.addAll(Arrays.asList(perks));
    }
}

File: src/main/kotlin/com/thebluehats/server/game/managers/world/PerkManager.kt
package com.thebluehats.server.game.managers.world

import com.thebluehats.server.game.perks.Perk
import com.thebluehats.server.game.utils.Registerer
import java.util.*

class PerkManager : Registerer<Perk> {
    private val perks = ArrayList<Perk>()

    override fun register(vararg objects: Perk) {
        this.perks.addAll(listOf(*objects))
    }
}